app/src/main/java/com/example/diceroller/Dice.java

```diff
@@ -0,0 +1,22 @@
+package com.example.diceroller;
+
+import java.util.Random;
+
+/**
+ * Dice with a fixed number of sides.
+ */
+class Dice {
+    private final Random random = new Random();
+    private final int numSides;
+
+    public Dice(int numSides) {
+        this.numSides = numSides;
+    }
+
+    /**
+     * Do a random dice roll and return the result.
+     */
+    public int roll() {
+        return random.nextInt(numSides) + 1;
```

Comment:
I think we should add a comment on the instructions on why this +1 is necessary given Java creates random numbers between 0 -> (n-1). Otherwise it might seem confusing for first time learners.

Reply:
Good callout! Are you suggesting something about it to the block comment above this method, or to the codelab, or both?

Reply:
I think a single line comment above this line of code could work! And then mentioning it in the codelab sounds good too. Thanks!

Reply:
Comment added in the code, and short explanation added in the codelab. 👍